tests/pmap_test.py

```diff
@@ -494,6 +494,66 @@ def sum_helper(a):
     ans = f(x)
     self.assertAllClose(ans, expected, check_dtypes=False)
 
+  def testGatherReplicaGroups(self):
```

Comment:
I would parameterize this test to run with two groups, to ensure that the case we fixed is working.

Reply:
I've added the case we discussed in the `testGatherReplicaGroupsInterleaved`. 
It's a bit tricky to generalize it into parameterized test case since the expected outputs need to be prepared differently.